### PR TITLE
3958 geocode all null or false

### PR DESCRIPTION
**Pull request by rafatower**
`cartodb_georef_status` semantics:
* NULL: not processed (before geocoding) or errored (a failure on our side)
* TRUE: succesful geocoding
* FALSE: failed geocoding (miss)

So at the beginning of the geocoding:
* formerly: all rows `WHERE cartodb_georef_status IS NULL` are to be processed
* now: all rows `WHERE cartodb_georef_status IS NOT TRUE` are to be processed

Changes:
* Add a column `force_all_rows` to the geocodings table
* Add parameter `force_all_rows` to create and `estimation_for` endpoints
* Modify `Geocoding#processable_rows` to take an optional param `force_all_rows` and account for it
* Rename `s/add_georef_status_column/ensure_georef_status_colummn_valid`
* Add a couple of methods `reset_cartodb_georef_status` and `mark_rows_to_geocode`
* Modify some of the sql expressions to keep consistent with everything

This covers #4407 (phase 1) AND phase 2 of #3958. @Kartones can you please review?

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4920/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4922/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4940/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4941/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4958/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4964/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4969/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4973/))

File: spec/models/geocoding_spec.rb [hunk 1]
```diff
@@ -142,6 +142,42 @@
...
+      before(:each) do
+        @my_table = FactoryGirl.create(:user_table, user_id: @user.id)
+      end
+
+      after(:each) do
+        @my_table.destroy
+      end
+
+      it 'marks rows to geocode with cartodb_georef_status = null' do
+        geocoding = FactoryGirl.build(:geocoding, user: @user, user_table: @my_table, kind: 'admin0', geometry_type: 'polygon', formatter: 'b')
```
Comment (Kartones) [inline spec/models/geocoding_spec.rb]: being lazy to check, that does `formatter: 'b'` do? (just curiosity)

**Comment by Kartones** [comment]
:+1:  nice!

File: spec/models/geocoding_spec.rb (hunk 1 above)
Comment (rafatower) [inline spec/models/geocoding_spec.rb]: `formatter` is a "polysemic" argument :(. For instance it contains `{street_column_name} {city_column_name} SomeCountry` for the hires geocoder but just `country_name_column` for admin regions geocoding. It's been like that for ages.

In the context of that test, not really interested in launching the actual geocoding but making sure before it is launched certain preconditions are met.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5002/))

**Pull request merged**